### PR TITLE
AP_Logger: make structure validation failure at runtime fatal

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

File: libraries/AP_Logger/AP_Logger.cpp [hunk 1]
```diff
@@ -871,7 +871,9 @@ AP_Logger::log_write_fmt *AP_Logger::msg_fmt_for_name(const char *name, const ch
     } else {
         memset((char*)ls_multipliers, '?', MIN(sizeof(ls_format), strlen(f->fmt)));
     }
-    validate_structure(&ls, (int16_t)-1);
+    if (!validate_structure(&ls, (int16_t)-1)) {
+        AP_HAL::panic("validation failed");
```
Comment (khancyr) [inline libraries/AP_Logger/AP_Logger.cpp]: log validation failed maybe ? 
Otherwise, just validation failed gives not clue on what is wrong

Comment (peterbarker) [reply libraries/AP_Logger/AP_Logger.cpp]: You've just been given a whole slew of stuff, so I figured this was OK :-)

Comment (khancyr) [reply libraries/AP_Logger/AP_Logger.cpp]: it is working at it is, but if you don't have a mavlink connexion open or printing correctly your text message, you will just have a generic "validation failed" ... that a little bit to generic as panic message

Comment (peterbarker) [reply libraries/AP_Logger/AP_Logger.cpp]: OK, changed it do do what the existing failure case did.

**Pull request merged**